File: 디자인패턴/java 디자인 패턴/PactoryMethod/ItemFactory.java
import java.util.HashMap;

public class ItemFactory extends Factory {
    private class ItemData {
        int maxCount, currentCount;
        ItemData(int maxCount) {
            this.maxCount = maxCount;
        }
    }

    private HashMap<String, ItemData> repository;

    public ItemFactory() {
        repository = new HashMap<String, ItemData>();
        repository.put("sword", new ItemData(3));
        repository.put("shield", new ItemData(2));
        repository.put("bow", new ItemData(1));
    }

	@Override
	public boolean isCreatable(String name) {
        ItemData itemData = repository.get(name);
        boolean Cheak;
        if(itemData == null) {
            System.out.println(name + "은 알 수 없는 아이템입니다.");
            Cheak = false;
        }
        else if(itemData.currentCount >= itemData.maxCount) {
            System.out.println(name + "은 품절 아이템입니다.");
            Cheak = false;
        }
        else Cheak = true;
        return Cheak;
	}

	@Override
	public Item createItem(String name) {
        Item item;
        if(name == "sword") item = new Sword();
        else if(name == "shield") item = new Shield();
        else if(name == "bow") item = new Bow();
        else {
            System.out.println("error! 잘못된 아이템 이름");
            return null;
        }
        return item;
	}

	@Override
	public void postProcessItem(String name) {
        if(name == "sword") {
            repository.get(name).currentCount++; }
        else if(name == "shield") {
            repository.get(name).currentCount++; }
        else if(name == "bow") {
            repository.get(name).currentCount++; }
        else {
            System.out.println("name error!"); }
	}
}

File: 디자인패턴/Python 디자인 패턴/PactoryMethod/ItemFactory.py
from Factory import*
from Item import Item
from ItemCreate import*

class ItemData():
        def __init__(this, maxCount:int) -> None:
            this.MaxCount = maxCount
            this.CurrentCount = 0
        pass

class ItemFactory(Factory):
    def __init__(this) -> None:
        this.Repository = {}
        this.Repository["sword"] = ItemData(maxCount=3)
        this.Repository["shield"] = ItemData(maxCount=2)
        this.Repository["bow"] = ItemData(maxCount=1)
    
    def isCreatable(this, name: str) -> bool:        
        if(this.Repository[name]==None):
            print(f"잘못된 아이템이름 {name}")
            itemCheck = False
        elif(this.Repository[name].CurrentCount 
                >= this.Repository[name].MaxCount):
            
            # TypeError: '>='은 'int'와 'ItemData' 
            # 인스턴스 간에는 지원되지 않습니다.
            print(f"재고없는 아이템이름 {name}")
            itemCheck = False
        else:itemCheck = True
        return itemCheck
    
    def CreateItem(this, name: str) -> Item:
        if(name=="sword"):item=Sword()
        elif(name=="shield"):item=Shield()
        elif(name=="bow"):item=Bow()
        else: 
            print(f"잘못된 아이템이름 {name}")
            return None
        return item
    
    def postProcessItem(this, name: str)->None:
        this.Repository[name].CurrentCount+=1
        pass
    pass